src/components/VForm/VForm.js

```diff
@@ -35,41 +30,80 @@ export default {
 
   methods: {
     getInputs () {
-      return this.$children.filter(child => {
-        return typeof child.errorBucket !== 'undefined'
+      const results = []
+
+      const search = (children, depth = 0) => {
+        for (const child of children) {
+          if (child.errorBucket !== undefined) {
+            results.push(child)
+          } else {
+            search(child.$children, depth + 1)
+          }
+        }
+        if (depth === 0) return results
+      }
+
+      return search(this.$children)
+    },
+    watchInputs (inputs) {
+      inputs === undefined && (inputs = this.getInputs())
+
+      for (const child of inputs) {
+        if (this.inputs.includes(child)) {
+          continue // We already know about this input
+        }
+
+        this.inputs.push(child)
+
+        this.watchChild(child)
+      }
+    },
+    watchChild (child) {
+      const watcher = (child) => {
+        child.$watch('valid', (val) => {
+          this.$set(this.errorBag, child._uid, !val)
+        }, { immediate: true })
+      }
+
+      if (!this.lazyValidation) return watcher(child)
+
+      // Only start watching inputs if we need to
+      child.$watch('shouldValidate', (val) => {
+        if (!val) return
+
+        // Only watch if we're not already doing it
+        if (this.errorBag.hasOwnProperty(child._uid)) return
+
+        watcher(child)
       })
     },
     validate () {
-      const errors = this.getInputs().reduce((errors, child) => {
-        const error = !child.validate(true)
-        return errors || error
-      }, false)
-
-      return !errors
+      return !this.inputs.filter(input => input.validate(true)).length
     },
     reset () {
-      this.getInputs().forEach((input) => input.reset())
+      this.inputs.forEach((input) => input.reset())
+      Object.keys(this.errorBag).forEach(key => this.$delete(this.errorBag, key))
     }
   },
 
   mounted () {
-    this.$vuetify.load(() => {
-      this.getInputs().forEach((child) => {
-        this.inputs += 1
+    this.$vuetify.load(() => this.watchInputs())
```

Comment:
This should be ```this.$vuetify.load(this.watchInputs)```

Reply:
That will pass the global `Window` object to `watchInputs`, instead of `undefined`. Is `.load()` even needed? Nothing seems to change if I remove it.

Reply:
It's for SSR protection lol

Reply:
Also valid point, no need to change this.